### PR TITLE
Update Documentation for Wrappers

### PR DESCRIPTION
**Pull request by hadleym**
All API functions now have parameter definitions and examples of usage.

### CODE REVIEW

**Comment by hadleym** [comment]
Also adds a python script for generating html, as well as implements a node module for generating html documentation for node wrapper.

File: vcx/wrappers/node/src/api/wallet.ts [hunk 1]
```diff
@@ -184,12 +185,16 @@ export class Wallet {
...
+   * Sends token to a specified address
+   *
+   * Example:
+   * ```
+   * address = await Wallet.createPaymentAddress('00000000000000000000000001234567')
+   * await Wallet.sendTokens({
+   *     payment: 0,
+   *     recipient: address,
+   *     tokens: 1
+   * }
```
Comment (thedolie) [inline vcx/wrappers/node/src/api/wallet.ts]: You are missing a closing `)` after the `}'

File: vcx/wrappers/node/src/api/wallet.ts [hunk 2]
```diff
@@ -217,9 +222,15 @@ export class Wallet {
...
-   * @description Adds a record to the wallet for storage
-   * @static
+   * Adds a record to the wallet for storage
+   * Example:
+   * ```
+   * await Wallet.addRecord({
+   *    id: 'RecordId',
+   *    tags: {},
+   *    type_: 'TestType',
+   *    value: 'RecordValue'})
```
Comment (thedolie) [inline vcx/wrappers/node/src/api/wallet.ts]: Generally, the paradigm for readability, at least in nodejs, is to have the closing brackets 1 line below the last parameter within it. It should close like this: 
await Wallet.addRecord({
       id: 'RecordId',
       tags: {},
       type_: 'TestType',
       value: 'RecordValue'
})

Comment (thedolie) [reply vcx/wrappers/node/src/api/wallet.ts]: This makes it much easier to reason about and youll find nearly all node code is written this way.

File: vcx/wrappers/node/src/api/wallet.ts [hunk 3]
```diff
@@ -254,12 +265,20 @@ export class Wallet {
...
+   * await Wallet.updateRecordValue({
+   *   id: 'RecordId',
+   *   type_: 'TestType',
+   *   value: 'RecordValueNew'})
```
Comment (thedolie) [inline vcx/wrappers/node/src/api/wallet.ts]: Newline

File: vcx/wrappers/node/src/api/wallet.ts [hunk 4]
```diff
@@ -291,12 +310,21 @@ export class Wallet {
...
-   * @returns {Promise<void>}
+   * Updates a record's tags already in the wallet
+   *
+   * Example:
+   * ```
+   * await Wallet.addRecord({
+   *    id: 'RecordId',
+   *    tags: {},
+   *    type_: 'TestType',
+   *    value: 'RecordValue'})
```
Comment (thedolie) [inline vcx/wrappers/node/src/api/wallet.ts]: new line

File: vcx/wrappers/node/src/api/wallet.ts [hunk 5]
```diff
@@ -291,12 +310,21 @@ export class Wallet {
...
+   *
+   * updateRecordTags({  id: 'RecordId',
+   *     tags: {},
+   *     type_: 'TestType',
+   *     value: ''})
```
Comment (thedolie) [inline vcx/wrappers/node/src/api/wallet.ts]: newline

File: vcx/wrappers/node/src/api/wallet.ts [hunk 6]
```diff
@@ -329,12 +357,21 @@ export class Wallet {
...
-   * @returns {Promise<void>}
+   * Adds tags to a record already in the wallet
+   *
+   * Example:
+   * ```
+   * await Wallet.addRecord({
+   *    id: 'RecordId',
+   *    tags: {},
+   *    type_: 'TestType',
+   *    value: 'RecordValue'})
```
Comment (thedolie) [inline vcx/wrappers/node/src/api/wallet.ts]: NewLine

File: vcx/wrappers/node/src/api/wallet.ts [hunk 7]
```diff
@@ -329,12 +357,21 @@ export class Wallet {
...
+   *
+   * addRecordTags({  id: 'RecordId',
+   *     tags: { "tagName1": "tag value 1", "~tagName2": "tag value 2 unencrypted", "tagName3", 1},
```
Comment (thedolie) [inline vcx/wrappers/node/src/api/wallet.ts]: Due to such a long line it should be formatted with each parameter being on its own line. Maybe this was auto generated?

File: vcx/wrappers/node/src/api/wallet.ts [hunk 8]
```diff
@@ -329,12 +357,21 @@ export class Wallet {
...
+   *     type_: 'TestType',
+   *     value: ''})
```
Comment (thedolie) [inline vcx/wrappers/node/src/api/wallet.ts]: Newline

**Pull request merged**